openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java

```diff
@@ -34,5 +42,77 @@ public DeploymentConfigOperationsImpl(OkHttpClient client, OpenShiftConfig confi
 
   public DeploymentConfigOperationsImpl(OkHttpClient client, OpenShiftConfig config, String apiVersion, String namespace, String name, Boolean cascading, DeploymentConfig item, String resourceVersion, Boolean reloadingFromServer, long gracePeriodSeconds, Map<String, String> labels, Map<String, String> labelsNot, Map<String, String[]> labelsIn, Map<String, String[]> labelsNotIn, Map<String, String> fields) {
     super(client, config, null, apiVersion, "deploymentconfigs", namespace, name, cascading, item, resourceVersion, reloadingFromServer, gracePeriodSeconds, labels, labelsNot, labelsIn, labelsNotIn, fields);
+    reaper = new DeploymentConfigReaper(this, client);
+  }
+
+  @Override
+  public DoneableDeploymentConfig edit() {
+    if (isCascading()) {
+      return cascading(false).edit();
+    }
+    return super.edit();
+  }
+
+  @Override
+  public DeploymentConfig replace(DeploymentConfig item) {
+    if (isCascading()) {
+      return cascading(false).replace(item);
+    }
+    return super.replace(item);
+  }
+
+  @Override
+  public DeploymentConfig patch(DeploymentConfig item) {
+    if (isCascading()) {
+      return cascading(false).patch(item);
+    }
+    return super.patch(item);
+  }
+
+  private static class DeploymentConfigReaper implements Reaper {
+
+    private final DeploymentConfigOperationsImpl operation;
+    private final OkHttpClient client;
+
+    public DeploymentConfigReaper(DeploymentConfigOperationsImpl operation, OkHttpClient client) {
+      this.operation = operation;
+      this.client = client;
+    }
+
+    @Override
+    public void reap() {
+      DeploymentConfig deployment = operation.cascading(false).edit().editSpec().withReplicas(0).endSpec().done();
+      waitForObservedGeneration(deployment.getStatus().getObservedGeneration());
```

Comment:
If we release kube-model & upgrade we should be able to work with available replicas - see https://github.com/fabric8io/kubernetes-model/blob/master/kubernetes-model/src/main/resources/schema/kube-schema.json#L8639-L8673